scripts/check_css.py

```
@@ -0,0 +1,181 @@
+
+class _StyleExtractor(HTMLParser):
+    """Collect the text content of every ``<style>`` element."""
+
+    def __init__(self) -> None:
+        """Initialise parser state."""
+        super().__init__()
+        self._in_style: bool = False
+        self._buf: list[str] = []
+        self.blocks: list[tuple[int, str]] = []  # (start_line, css_text)
+
+    def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None:
```

Ruff (0.15.5)

[warning] Unused method argument: `attrs`

(ARG002)